workspace/src/localization/particle_filter_estimation/particle_filter_estimation/particle_filter_estimation.py

```diff
@@ -0,0 +1,205 @@
+import csv
+import rclpy
+from rclpy.node import Node
+from art_msgs.msg import VehicleState
+from chrono_ros_interfaces.msg import DriverInputs as VehicleInput
+from sensor_msgs.msg import Imu, NavSatFix, MagneticField
+from chrono_ros_interfaces.msg import Body as ChVehicle
+import matplotlib.pyplot as plt
+import matplotlib
+import math
+import numpy as np
+import sys
+import os
+from enum import Enum
+from localization_py.particle_filter import ParticleFilter as PF
+from localization_shared_utils import get_dynamics, get_coordinate_transfer
+
+
+class ParticleFilterEstimationNode(Node):
+    def __init__(self):
+        super().__init__("particle_filter_estimation_node")
+
+        # ROS PARAMETERS
+        self.use_sim_msg = (
+            self.get_parameter("use_sim_time").get_parameter_value().bool_value
+        )
+
+        # dynamics parameters
+        self.declare_parameter("c_1", 0.0001)
+        c_1 = self.get_parameter("c_1").get_parameter_value().double_value
+        self.declare_parameter("c_0", 0.02)
+        c_0 = self.get_parameter("c_0").get_parameter_value().double_value
+        self.declare_parameter("l", 0.5)
+        l = self.get_parameter("l").get_parameter_value().double_value
+        self.declare_parameter("r_wheel", 0.08451952624)
+        r_wheel = self.get_parameter("r_wheel").get_parameter_value().double_value
+        self.declare_parameter("i_wheel", 0.001)
+        i_wheel = self.get_parameter("i_wheel").get_parameter_value().double_value
+        self.declare_parameter("gamma", 0.33333333)
+        gamma = self.get_parameter("gamma").get_parameter_value().double_value
+        self.declare_parameter("tau_0", 0.3)
+        tau_0 = self.get_parameter("tau_0").get_parameter_value().double_value
+        self.declare_parameter("omega_0", 30.0)
+        omega_0 = self.get_parameter("omega_0").get_parameter_value().double_value
+        dyn = [c_1, c_0, l, r_wheel, i_wheel, gamma, tau_0, omega_0]
+        self.get_logger().info(str(dyn))
+
+        # update frequency of this node
+        self.freq = 10.0
+
+        self.gps = ""
+        self.ground_truth = ""
+        self.mag = ""
+
+        # x, y, from measurements
+        self.x = 0
+        self.y = 0
+
+        # what we will be using for our state vector. (x, y, theta yaw, v vel)
+        self.state = np.zeros((4, 1))
+        np.vstack(self.state)
```

Comment:
vstack isn't inline, it will return the stacked state. This line doesn't actually do anything then.